java/client/src/org/openqa/selenium/devtools/profiler/model/ConsoleProfileFinished.java

```diff
@@ -0,0 +1,105 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.selenium.devtools.profiler.model;
+
+import java.util.Objects;
+import org.openqa.selenium.json.JsonInput;
+
+public class ConsoleProfileFinished {
+
+  private String id;
+  /**
+   * Location of console.profileEnd().
+   */
+  private Location location;
+
+  private Profile profile;
+
+  /**
+   * Profile title passed as an argument to console.profile().
+   *
+   * <p>Optional
+   */
+  private String title;
+
+  public ConsoleProfileFinished(String id, Location location, Profile profile, String title) {
```

Comment:
Check that all non-optional fields are `nonNull` in the constructor, so we can never have instances in an illegal state.

Reply:
@shs96c I have done the validation via Setters , if it's prefer to be done via the constructor I will fix it.